### PR TITLE
Fix build hanging

### PR DESCRIPTION
**Pull request by orhtej2**
## Problem

- Install hanging due to change to `corepack`
- `yarn` classic installed despite upstream using 4.x
- `bookworm` install failing

## Solution

- Properly activate `corepack`
- Use `yarn` from `corepack`
- Properly populate `env`
- Make `node` use swap space

## PR Status

- [x] Code finished and ready to be reviewed/tested
- [x] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by orhtej2** [comment]
!testme

**Comment by orhtej2** [comment]
!bookwormtestme

**Comment by orhtej2** [comment]
!testme

**Comment by orhtej2** [comment]
!testme

**Comment by orhtej2** [comment]
!testme

**Comment by orhtej2** [comment]
I mean it passed `bookworm` build on my machine, some resource starvation on CI?

File: scripts/install [hunk 1]
```diff
@@ -122,22 +122,24 @@ pushd "$install_dir/live"
...
+	ynh_exec_as "$app" env $ynh_ruby_load_path $ld_preload bin/bundle config without 'development test'
+	ynh_exec_as "$app" env $ynh_ruby_load_path $ld_preload bin/bundle config set force_ruby_platform true
+	ynh_exec_as "$app" env $ynh_ruby_load_path $ld_preload bin/bundle install -j$(getconf _NPROCESSORS_ONLN)
 
 	# Building assets
 	ynh_use_nodejs
-	corepack enable
-	yarn install
+	env $ynh_node_load_PATH corepack enable
+	ynh_exec_warn_less ynh_exec_as "$app" env $ynh_node_load_PATH corepack enable
```
Comment (Salamandar) [inline scripts/install:133]: Hmmm why line 132 ? wouldn't line 133 be enough ?

Comment (orhtej2) [reply scripts/install:133]: If you run as a non-admin user first it'll try to links something in global scope and fall flat on it's face 🤷

Comment (Salamandar) [reply scripts/install:133]: Yeah indeed.
I got a less impactful fix with an exported env var, in the current pacaging v2 branch, but i like your approach and will test this after packagingv2 is merged onto testing.

**Comment by ericgaspar** [comment]
!testme

**Comment by orhtej2** [comment]
`master` now uses alternative fix for the problem, we can get rid of this.

**Pull request closed without merging**